ui/app/pages/index.us

```diff
@@ -5,6 +5,18 @@
     <meta charset="utf-8">
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
+    <!--
+        Content-Security-Policy explanation:
+        * img-src: gravatar & githubusercontent for user avatars
+        * frame-src: *.auth0.com to allow bearer token renewals in the background
+        * connect-src: *.auth0.com to allow us to pull public keys from auth0 to validate the login
+        * script-src: unsafe-{eval,inline} because some of our dependencies require it
+        * style-src: unsafe-inline because some of our dependencies require it. fonts.googleapis.com, because we load a font from it (in a weird way that injects it to the app at runtime)
+        * font-src: fonts.googleapis.com because we load a font from it
+        * frame-ancestors: none because there's no valid reason to embed Balrog in another page
+    -->
+    <meta http-equiv="Content-Security-Policy"
+          content="default-src 'none'; img-src 'self' *.gravatar.com *.githubusercontent.com; frame-src https://*.auth0.com/; connect-src 'self' https://*.auth0.com/; script-src 'self' 'unsafe-eval' 'unsafe-inline'; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com/; font-src 'self' https://fonts.gstatic.com">
```

Comment:
Assuming there are TLS versions of the image sources I think we could do: `https://*.gravatar.com https://*.githubusercontent.com`

Also, I'd be curious about which dependencies need unsafe scripts sources for future deprecation.

Overall looks good!

Reply:
> Assuming there are TLS versions of the image sources I think we could do: https://*.gravatar.com https://*.githubusercontent.com

Hmmm, let me see if this will work for us...I think it's the auth0 code that pulls them, so it may not...

> Also, I'd be curious about which dependencies need unsafe scripts sources for future deprecation.

To be honest, I haven't been able to figure out how to trace it down. However, because we run a super ancient angular and node stack, we have some super ancient dependencies which may be part of it. This UI is going away before EOY though, and I expect we'll be able to lock this down further when this one dies.

Reply:
> > Assuming there are TLS versions of the image sources I think we could do: https://_.gravatar.com https://_.githubusercontent.com
> 
> Hmmm, let me see if this will work for us...I think it's the auth0 code that pulls them, so it may not...

Setting https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/upgrade-insecure-requests might handle that; I'm not sure if that is applied before checking against the allowlist here.